third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -641,6 +762,19 @@ void Pingponger::getDotPingponged() {
     return;
   }
 
```

Comment:
Sorry, I was bit distracted by several other tasks.
The logic looks good to me, just want to check I understand correctly.
So, basically we don't allow any estimateNonDotMemoryImpact<*> here?
Then this might look simpler.

```
//split dot/non-dot memory ops
stable_partition gLoadOps
stable_partition lLoadOps
stable_partition lStoreOps

// reject if any effective mem ops in non-dot memory ops
if (estimateNonDotMemoryImpact<*> != 0)
  return 

// delete non-dot memory ops from movable ops
gLoadOps.erase
lLoadOps.erase
lStoreOps.erase

// final check
if (gLoadOps.size() != 2 || lLoadOps.size() != 2) {
  return
```

Reply:
Sure I can cleanup this code to be simplified as you suggested. Thanks!